### PR TITLE
feat: switch from netmask to ipaddr.js for ipv6 support COMPASS-6795

### PR DESCRIPTION
**Pull request by lerouxb**
https://github.com/whitequark/ipaddr.js

I also now changed `npm run prepare` so that it writes a cidrs.js file with everything pre-parsed. This means that it won't block at runtime. The downside is that we have the lists baked in and have to regularly re-build, but that was already the case because the azure lists were from 2019 and it doesn't look like there a URL where we can easily get the up-to-date list automatically.

### CODE REVIEW

File: test/index.spec.js [hunk 1]
```diff
@@ -72,5 +71,5 @@ describe('getCloudInfo', () => {
       isAzure: true
     });
   });
-});
+}).timeout(5000);
```
Comment (lerouxb) [inline test/index.spec.js:74]: Our fetch timeout for AWS ranges is 3 seconds which is already higher than the default mocha timeout.

File: index.js [hunk 2]
```diff
@@ -1,34 +1,26 @@
...
   const lookup = util.promisify(gceIpsInstance.lookup.bind(gceIpsInstance));
 
-  return (await lookup())
-    .filter(isIpv4Range);
+  return await lookup();
 }
 
 async function getAwsIpRanges() {
   const awsIpRangesUrl = 'https://ip-ranges.amazonaws.com/ip-ranges.json';
   const { prefixes } = await fetch(awsIpRangesUrl, { timeout: 3000 }).then(res => res.json());
```
Comment (gribnoysup) [inline index.js:20]: There are ipv6_prefixes key in this object, we probably want that too now that we check them?

File: index.js [hunk 3]
```diff
@@ -41,12 +33,19 @@ async function getAzureIpRanges() {
...
 function rangeContainsIp(ipRanges, ip) {
-  return !!ipRanges.find((cidr) => new Netmask(cidr).contains(ip));
+  const address = ipaddr.parse(ip);
+  return !!ipRanges.find((ipRange) => {
+    const cidr = ipaddr.parseCIDR(ipRange);
+    if (address.kind() !== cidr[0].kind()) {
+      // cannot match ipv4 address with non-ipv4 one
+      return false;
+    }
+    return address.match(cidr);
```
Comment (gribnoysup) [inline index.js:47]: Maybe saw this, but this is basically `ipaddr.subnetMatch`

File: index.js [hunk 4]
```diff
@@ -41,12 +33,19 @@ async function getAzureIpRanges() {
...
     .values
     .map(value => value.properties.addressPrefixes)
-    .reduce((acc, val) => acc.concat(val), [])
-    .filter(isIpv4Range);
+    .reduce((acc, val) => acc.concat(val), []);
 }
 
 function rangeContainsIp(ipRanges, ip) {
-  return !!ipRanges.find((cidr) => new Netmask(cidr).contains(ip));
+  const address = ipaddr.parse(ip);
```
[the remark this answers is not in the document]
Comment (lerouxb) [reply index.js:40]: I know. But we do that three times total vs 16420+7432+58 for the ranges which is why I haven't bothered yet :)

**Comment by gribnoysup** [comment]
If we are updating this, we probably can pull a new [azure ip addresses file](https://www.microsoft.com/en-us/Download/details.aspx?id=56519) in?

File: index.js (hunk 2 above)
Comment (lerouxb) [reply index.js:20]: Ooh I didn't notice that. Fixing.

File: index.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (lerouxb) [reply index.js:40]: 9286 rather than 7432 once we properly use the ipv6 AWS list as well

[the remark this answers is not in the document]
Comment (gribnoysup) [reply index.js:40]: Yeah, that's fair 😄

[the remark this answers is not in the document]
Comment (lerouxb) [reply index.js]: I optimised it now anyway ;)

File: .github/workflows/update-cidrs.yaml [hunk 5]
```diff
@@ -0,0 +1,60 @@
+name: Update cidrs.json
+
+on:
+  # Once a week or on pushes to main
+  schedule:
+    - cron: "0 3 * * 0"
```
Comment (mcasimir) [inline .github/workflows/update-cidrs.yaml:6]: we can do hourly too

File: .github/workflows/update-cidrs.yaml [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+  push:
+    branches:
+      - main
```
Comment (mcasimir) [inline .github/workflows/update-cidrs.yaml:9]: probably better to replace this with "workflow_dispatch" so we have a way to trigger it manually

**Review by gribnoysup — APPROVED** [review]

File: .github/workflows/update-cidrs.yaml (hunk 5 above)
Comment (lerouxb) [reply .github/workflows/update-cidrs.yaml]: changed to nightly

**Pull request merged**